test/extended/authorization/podsecurity_admission.go

```diff
@@ -189,14 +193,302 @@ var _ = g.Describe("[sig-auth][Feature:PodSecurity][Feature:SCC]", func() {
+						o.Expect(ns.GetLabels()[psapi.WarnLevelLabel]).To(o.Equal(psaLabel))
+					} else {
+						// SCC assignment does not yield any PSa label change
+						o.Expect(ns.GetLabels()[psapi.EnforceLevelLabel]).To(o.Equal(testCase.expectedPSaLevelAfterUpdate))
+						o.Expect(ns.GetLabels()[psapi.AuditLevelLabel]).To(o.Equal(testCase.expectedPSaLevelAfterUpdate))
+						o.Expect(ns.GetLabels()[psapi.WarnLevelLabel]).To(o.Equal(testCase.expectedPSaLevelAfterUpdate))
+					}
+				}
+			})
+		}
+	})
+})
+
+// creates a role & rolebinding for the test and binds it to the default SA
+func bindRoleToDefaultSA(ctx context.Context, oc *exutil.CLI, nsName string) (*v1.Role, error) {
+	role := &v1.Role{
+		ObjectMeta: metav1.ObjectMeta{Name: "my-role-with-scc", Namespace: nsName},
+	}
+
+	roleBinding := &v1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{Name: "my-role-binding", Namespace: nsName},
+		Subjects: []v1.Subject{
+			{Kind: v1.ServiceAccountKind, Name: "default"},
+		},
+		RoleRef: v1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     role.ObjectMeta.Name,
+		},
+	}
+
+	_, err := oc.AdminKubeClient().RbacV1().Roles(nsName).Create(ctx, role, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = oc.AdminKubeClient().RbacV1().RoleBindings(nsName).Create(ctx, roleBinding, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return role, nil
+}
+
+// updates a role to use a specific SCC
+func setSCCPermissions(ctx context.Context, oc *exutil.CLI, role *v1.Role, scc string) error {
+	role.Rules = []v1.PolicyRule{
+		{
+			APIGroups:     []string{"security.openshift.io"},
+			ResourceNames: []string{scc},
+			Resources:     []string{"securitycontextconstraints"},
+			Verbs:         []string{"use"},
+		},
+	}
+
+	_, err := oc.AdminKubeClient().RbacV1().Roles(role.ObjectMeta.Namespace).Update(ctx, role, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// creates or updates a namespace based on the options passed, then waits a bit for any controllers to apply
+// changes and returns an up-to-date namespace object
+func createOrUpdateNamespace(ctx context.Context, oc *exutil.CLI, ns *corev1.Namespace, options interface{}) (*corev1.Namespace, error) {
+	var err error
+	switch options.(type) {
+	case metav1.CreateOptions:
+		_, err = oc.AdminKubeClient().CoreV1().Namespaces().Create(ctx, ns, options.(metav1.CreateOptions))
+		oc.AddExplicitResourceToDelete(corev1.SchemeGroupVersion.WithResource("namespaces"), "", ns.Name)
+	case metav1.UpdateOptions:
+		_, err = oc.AdminKubeClient().CoreV1().Namespaces().Update(ctx, ns, options.(metav1.UpdateOptions))
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	ns, err = waitAndGetNamespace(ctx, oc, ns.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	return ns, nil
+}
+
+func waitAndGetNamespace(ctx context.Context, oc *exutil.CLI, nsName string) (*corev1.Namespace, error) {
```

Comment:
you don't need this, once an object is created, it's created

Reply:
Isn't he giving the pod security label syncer time to act?